Common/Data/SubscriptionDataSource.cs

```diff
@@ -44,24 +44,33 @@ public class SubscriptionDataSource : IEquatable<SubscriptionDataSource>
         /// </summary>
         public readonly IReadOnlyList<KeyValuePair<string, string>> Headers;
 
+        /// <summary>
+        /// Flags if algorithm is running on a live server.
+        /// </summary>
+        public readonly bool? IsLiveMode;
```

Comment:
@Jay-Jay-D -- I don't think we need this here. The individual subscription factories will know whether or not they're in live mode or not. Ideally the `SubscriptionDataSource` doesn't care what type of algorithm we're running. Let's remove this and source the `isLiveMode` value elsewhere. Ping me on slack if you get caught in any particular spot, but IIRC, the data should already be available where it's needed